### PR TITLE
NEW_OFFSET_CONSISTENCY_SUM

### PR DESCRIPTION
**Pull request by Nikita-Katyshev**
## What
`NEW_OFFSET_CONSISTENCY_SUM` query is introduced. It's not used in a VisibilityCheck but is logged.

## Why
To prove it can be used as an `OFFSET_CONSISTENCY_SUM`.

## How
both `OFFSET_CONSISTENCY_SUM` and `NEW_OFFSET_CONSISTENCY_SUM` are executed, but only `OFFSET_CONSISTENCY_SUM` is used as a result. `NEW_OFFSET_CONSISTENCY_SUM` is logged to verify it has equal value for the tills for which `OFFSET_CONSISTENCY_SUM` gives the equal value.

### CODE REVIEW

**Review by NPhalgunaRao — APPROVED** [review]
Changes looks good

**Review by akshkmr — APPROVED** [review]
LGTM

File: pipe-storage-sqlite/src/main/java/com/tesco/aqueduct/pipe/storage/sqlite/SQLiteStorage.java [hunk 1]
```diff
@@ -486,10 +486,16 @@ private void reindex(Connection connection) {
...
 
     private long getOffsetConsistencySumBasedOn(long offsetThreshold, Connection connection) throws SQLException {
-        try (PreparedStatement statement = connection.prepareStatement(SQLiteQueries.OFFSET_CONSISTENCY_SUM)) {
+        try (
+                PreparedStatement statement = connection.prepareStatement(SQLiteQueries.OFFSET_CONSISTENCY_SUM);
+                PreparedStatement newStatement = connection.prepareStatement(SQLiteQueries.NEW_OFFSET_CONSISTENCY_SUM)
+        ) {
             statement.setLong(1, offsetThreshold);
             statement.setLong(2, offsetThreshold);
-            return queryResult(statement);
```
Comment (NPhalgunaRao) [inline pipe-storage-sqlite/src/main/java/com/tesco/aqueduct/pipe/storage/sqlite/SQLiteStorage.java:492]: I think you have to add below line 

newStatement.setLong(1, offsetThreshold);

**Review by ryzhovart — DISMISSED** [review]

File: pipe-storage-sqlite/src/main/java/com/tesco/aqueduct/pipe/storage/sqlite/SQLiteStorage.java [hunk 2]
```diff
@@ -486,10 +486,16 @@ private void reindex(Connection connection) {
...
+            long offsetConsistencySum = queryResult(statement);
+            long newOffsetConsistencySum = queryResult(newStatement);
+            LOG.info("getOffsetConsistencySumBasedOn", "newOffsetConsistencySum=" + newOffsetConsistencySum);
+            return offsetConsistencySum;
```
Comment (GriIvanov) [inline pipe-storage-sqlite/src/main/java/com/tesco/aqueduct/pipe/storage/sqlite/SQLiteStorage.java:498]: No metric push needed here?

Comment (Nikita-Katyshev) [reply pipe-storage-sqlite/src/main/java/com/tesco/aqueduct/pipe/storage/sqlite/SQLiteStorage.java:498]: We have Splunk logs for now, so it should work. We do not want to keep it for a long time.
metrics may be pushed but it's not required in my opinion.

File: pipe-storage-sqlite/src/main/java/com/tesco/aqueduct/pipe/storage/sqlite/SQLiteStorage.java (hunk 1 above)
Comment (Nikita-Katyshev) [reply pipe-storage-sqlite/src/main/java/com/tesco/aqueduct/pipe/storage/sqlite/SQLiteStorage.java:492]: good catch, thanks.
fixed

**Review by akshkmr — APPROVED** [review]
LGTM

**Review by NPhalgunaRao — APPROVED** [review]

**Review by NPhalgunaRao — APPROVED** [review]

**Pull request merged**